include/sys/trace.h

```diff
@@ -43,19 +43,23 @@
  */
 #undef DTRACE_PROBE1
 #define	DTRACE_PROBE1(name, t1, arg1) \
-	trace_zfs_##name((arg1))
+	if (arg1) \
+		trace_zfs_##name((arg1))
 
 #undef DTRACE_PROBE2
 #define	DTRACE_PROBE2(name, t1, arg1, t2, arg2) \
-	trace_zfs_##name((arg1), (arg2))
+	if (arg1 && arg2) \
```

Comment:
There's a good LWN article describing how to use tracepoints which I'd recommend.  It provides a nice summary including some examples of what's possible.

  https://lwn.net/Articles/379903/

After a little digging I could not find an example of a case where a tracepoint argument is NULL.  Usually this is handled by defining another tracepoint which only takes the required number of arguments.  For example.

```diff
diff --git a/module/zfs/dbuf.c b/module/zfs/dbuf.c
index dc2c004..eea209b 100644
--- a/module/zfs/dbuf.c
+++ b/module/zfs/dbuf.c
@@ -508,16 +508,19 @@ dbuf_evict_one(void)
                db = multilist_sublist_prev(mls, db);
        }
 
-       DTRACE_PROBE2(dbuf__evict__one, dmu_buf_impl_t *, db,
-           multilist_sublist_t *, mls);
-
        if (db != NULL) {
+               DTRACE_PROBE2(dbuf__evict__one, dmu_buf_impl_t *, db,
+                   multilist_sublist_t *, mls);
+
                multilist_sublist_remove(mls, db);
                multilist_sublist_unlock(mls);
                (void) refcount_remove_many(&dbuf_cache_size,
                    db->db.db_size, db);
                dbuf_destroy(db);
        } else {
+               DTRACE_PROBE1(dbuf__evict__one_mls,
+                   multilist_sublist_t *, mls);
+
                multilist_sublist_unlock(mls);
        }
        (void) tsd_set(zfs_dbuf_evict_key, NULL);
```

That may be what we have to do here even if it means diverging slightly from upstream.  It might be possible to use the same conditional logic for printing custom fields on the arguments themselves in the `dbuf__evict__one` definition but I haven't tried if that works.